### PR TITLE
create tournament screen, some tournament requests

### PR DESCRIPTION
**Pull request by williambertrand**
### Create Tournament Screen & some requests
- Trying to keep this PR from growing in scope too much, so for now, literally just show the Create page, enter name, and submit request.
- Adds the new post/put/get requests
- Will do Tournament Details page in another PR

### TODOs

- [x] Merge in the work from #4 to un-error a few things
- [x] add a simple test for this screen
- [x] Navigate from home to Create

| Capture |
| --- |
|  |

### CODE REVIEW

File: screens/TournamentCreation/__tests__/CreateTournamentScreen.test.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+describe("CreateTournamentScreen", () => {
+    test("it renders header and name input", () => {
+        const { getByText, getByRole } = render(<CreateTournamentScreen />);
+        expect(getByText("Create Tournament")).toBeDefined();
+        expect(getByText("Tournament Name")).toBeDefined();
+        
+        // Create Button
+        expect(getByRole("button")).toBeDefined();
+    });
+})
```
Comment (williambertrand) [inline screens/TournamentCreation/__tests__/CreateTournamentScreen.test.tsx:15]: Example of a very simple test - literally just "if someone breaks this component, it should fail"

File: screens/RegisterScreen.tsx [hunk 2]
```diff
@@ -9,29 +9,25 @@ import {
...
-        if (json.success) {
-          navigation.navigate("Login");
-        }
-      },
-    });
+  async function register() {
+    const body = {
+      username,
+      password,
+      email,
```
Comment (williambertrand) [inline screens/RegisterScreen.tsx:25]: When the object name is the same as the object value, you can shorthand like this

File: screens/RegisterScreen.tsx [hunk 3]
```diff
@@ -9,29 +9,25 @@ import {
...
-      email: email,
-    });
-    postRequest({
-      path: emailRegisterPath,
-      body: body,
-      callback: (json: { success: boolean }) => {
-        if (json.success) {
-          navigation.navigate("Login");
-        }
-      },
```
Comment (williambertrand) [inline screens/RegisterScreen.tsx:33]: Avoid callbacks at all costs, `await` for promise to resolve instead

File: screens/TournamentCreation/__tests__/CreateTournamentScreen.test.tsx (hunk 1 above)
Comment (probicheaux) [reply screens/TournamentCreation/__tests__/CreateTournamentScreen.test.tsx:15]: Cool, yeah, I have no idea how tests work in js/ts so I'll have to look into what the standard stuff is mroe

**Review by probicheaux — APPROVED** [review]
Nice, LGTM. Gonna get working on getting the api wired up, lmk if you or I should do the firebase auth listening stuff, I'm down for either.

**Pull request merged**